Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -2012,7 +2012,28 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
      */
     private func register(_ bundle: DocumentationBundle) throws {
         try shouldContinueRegistration()
-        
+
+        let currentFeatureFlags: FeatureFlags?
+        if let bundleFlags = bundle.info.featureFlags {
+            currentFeatureFlags = FeatureFlags.current
+            FeatureFlags.current.loadFlagsFromBundle(bundleFlags)
+
+            for unknownFeatureFlag in bundleFlags.unknownFeatureFlags {
+                diagnosticEngine.emit(.init(diagnostic:
+                        .init(
+                            severity: .warning,
+                            identifier: "org.swift.docc.UnknownBundleFeatureFlag",
+                            summary: "Unknown feature flag in Info.plist: \(unknownFeatureFlag.singleQuoted)")))
+            }
+        } else {
+            currentFeatureFlags = nil
+        }
+        defer {
+            if let currentFeatureFlags = currentFeatureFlags {
```

Comment:
This is a nice touch, but in practice do we ever compile more than one bundle in the same DocC invocation? Actually I just read @d-ronnqvist 's comment from last week... but this is a good idea, especially for unit tests maybe.

Reply:
Right, we may not really support building more than one bundle in the same invocation, but i still wanted the behavior to be temporary for unit tests.